lib/-private/file-path.js

```diff
@@ -0,0 +1,73 @@
+import { parse } from 'node:path';
+
+const DTS = '.d.ts';
+export const GJS = ['.gjs', '.gts'];
+const HBS = '.hbs';
+const HTML = '.html';
+export const SOON_DEPRECATED = ['.js', '.ts'];
+
+function isDTS(ext) {
+  return ext.endsWith(DTS);
+}
+
+function isGlimmerScript(ext) {
+  return GJS.includes(ext);
+}
+
+function isHBS(ext) {
+  return ext === HBS;
+}
+
+function isHTML(ext) {
+  return ext === HTML;
+}
+
+function isScript(ext) {
+  return SOON_DEPRECATED.includes(ext);
+}
+
+export function canProcessFile(filePath) {
+  if (!filePath) {
+    return false;
+  }
+
+  const { ext } = parseFilePath(filePath);
+
+  return isDTS(ext) || isGlimmerScript(ext) || isHBS(ext) || isHTML(ext) || isScript(ext);
+}
+
+export function parseFilePath(filePath) {
```

Comment:
Code tested in [`@codemod-utils/files`](https://github.com/ijlee2/codemod-utils/blob/2.3.2/packages/files/src/files/parse-file-path.ts). `filePath` is assumed to be well-defined (has the type of `string`).